### DISCUSSION TITLE
it was working and now it does not // objection

### DISCUSSION THREAD

**Discussion by VAROME** [vfsfitvnm/frida-il2cpp-bridge#387] [Q&A]
i once did this 
- created working tested JS script that changes game currency 
- used orignal apk 
- confg file
```
{
  "interaction": {
    "type": "script",
    "path": "libfrida-gadget.script.so",
    "on_load": "resume"
  }
}
```
then 

`objection patchapk -s apk.apk -c config.json -l index.js --architecture armeabi-v7a
`
after that to make it on non rooted i did
config2 

```
{
  "interaction": {
    "type": "listen",
    "address": "127.0.0.1",
    "port": 27042,
    "on_load": "resume"
  }
}
```

and then 
`objection patchapk -s apk.objection.apk -c config2.json --architecture armeabi-v7a
`
then it was working , 
now it does not , even i tried a simple working one and all
so how to make a non rooted apk with that script i created ? since objection does not really work now !

- one more thing , 
- the objecton thing, after that the game runs , but not working the script itself , but if i open the gadget frida thing from pc i inject normally the script from pc and runs normally , but not the one that has already been injected, so objection makes the apk run gadget without pausing but at the same time it does not run the script . yet accepts from the PC any script again , or any code to run like without objection

**Comment by bluewave41** [comment]
Objection works fine. I use it for a project and there are no issues.

This will be something wrong with your configuration or build steps. I'd assume you're probably using frida-compile which has a number of issues since it moved to using ESM modules. If you are, ditch that for esbuild.

Besides that there's nothing else to tell you without code.

**Reply by VAROME** [reply]
the objection works fine yes, but some of the functions that are called like a menu is working fine while gadget and adb connected , the same code would just not work when objection used , like menu not showing up , beside other functions works like a menu toast from the same code , it is odd . 
give me a communication way , my code is private i really worked hard on it and not ready to publish it yet. ,

**Reply by christiancoleman** [reply]
How does one switch from using frida-compile to esbuild? I think frida-compile may be the root of most of my build problems.

**Reply by IrvanWijayaSardam** [reply]
i'm facing the same issue right now, would you please help me how to solve it ?

**Comment by VAROME** [comment]
close

**Comment by VAROME** [comment]
easy , with a switch , xD 😉
just google it you would find the esbuild code , use it instead the frida
compile
and install esbuild on vscode

**Reply by forforgebe** [reply]
hello, how did you solve the issue with mod menu. I had same issue. I already built my script with esbuild. But only work if I injected the script with adb.
